receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -124,21 +124,50 @@ func (t *transaction) Append(ref storage.SeriesRef, ls labels.Labels, atMs int64
 		return 0, t.AddTargetInfo(ls)
 	}
 
-	curMF, ok := t.families[metricName]
+	curMF := t.getOrCreateMetricFamily(metricName)
+
+	return 0, curMF.Add(metricName, ls, atMs, val)
+}
+
+func (t *transaction) getOrCreateMetricFamily(mn string) *metricFamily {
+	curMf, ok := t.families[mn]
 	if !ok {
-		familyName := normalizeMetricName(metricName)
-		if mf, ok := t.families[familyName]; ok && mf.includesMetric(metricName) {
-			curMF = mf
+		fn := normalizeMetricName(mn)
+		if mf, ok := t.families[fn]; ok && mf.includesMetric(mn) {
+			curMf = mf
 		} else {
-			curMF = newMetricFamily(metricName, t.mc, t.logger)
-			t.families[curMF.name] = curMF
+			curMf = newMetricFamily(mn, t.mc, t.logger)
+			t.families[curMf.name] = curMf
 		}
 	}
-
-	return 0, curMF.Add(metricName, ls, atMs, val)
+	return curMf
 }
 
 func (t *transaction) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e exemplar.Exemplar) (storage.SeriesRef, error) {
+	select {
+	case <-t.ctx.Done():
+		return 0, errTransactionAborted
+	default:
+	}
+
+	if t.isNew {
+		if err := t.initTransaction(l); err != nil {
+			return 0, err
+		}
+	}
+
```

Comment:
Based on the prometheus server's [implementation](https://github.com/prometheus/prometheus/blob/4927e1353786c89e847e39c41a05a6f8ee464465/tsdb/head_append.go#L371), we should filter out empty labels using `WithoutEmpty()`

Reply:
Done